### PR TITLE
[dbus] fix dbus reconnection

### PR DESCRIPTION
**Pull request by simonlingoogle**
Found that `dbus_bus_request_name` may fail on Raspberry Pi. 

This commit fixes the issue by re-connecting DBus in such case.

### CODE REVIEW

File: src/dbus/server/dbus_agent.cpp [hunk 1]
```diff
@@ -49,47 +49,58 @@ DBusAgent::DBusAgent(otbr::Ncp::ControllerOpenThread &aNcp)
...
-    DBusError   dbusError;
-    otbrError   error = OTBR_ERROR_NONE;
-    int         requestReply;
-    std::string serverName = OTBR_DBUS_SERVER_PREFIX + mInterfaceName;
+    otbrError error = OTBR_ERROR_NONE;
 
-    dbus_error_init(&dbusError);
-    DBusConnection *conn                = nullptr;
-    auto            connection_deadline = std::chrono::steady_clock::now() + std::chrono::seconds(30);
+    auto connection_deadline = std::chrono::steady_clock::now() + std::chrono::seconds(30);
```
Comment (wgtdkp) [inline src/dbus/server/dbus_agent.cpp:54]:
```suggestion
    auto connection_deadline = Clock::now() + Seconds(30);
```

consider using the `Clock` type alias in `common/time.hpp` to enforce always using `steady_clock`.
```c++
using Seconds      = std::chrono::seconds;
using Milliseconds = std::chrono::milliseconds;
using Microseconds = std::chrono::microseconds;
using Clock        = std::chrono::steady_clock;
using Timepoint    = Clock::time_point;
```

Comment (simonlingoogle) [reply src/dbus/server/dbus_agent.cpp]: Added `using Clock        = std::chrono::steady_clock;`

But other suggestions are not accepted.

**Review by wgtdkp — APPROVED** [review]

**Review by jinran-google — APPROVED** [review]
LGTM :+1:

**Review by bukepo — APPROVED** [review]

File: src/dbus/server/dbus_agent.cpp [hunk 2]
```diff
@@ -49,47 +51,58 @@ DBusAgent::DBusAgent(otbr::Ncp::ControllerOpenThread &aNcp)
...
-        {
-            break;
-        }
-        otbrLogWarning("Failed to get DBus connection: %s: %s, will retry after 3 seconds", dbusError.name,
-                       dbusError.message);
-        dbus_error_free(&dbusError);
-        sleep(3);
+        otbrLogWarning("Failed to setup DBus connection, will retry after 1 second");
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+        mConnection = PrepareDBusConnection();
```
Comment (jwhui) [inline src/dbus/server/dbus_agent.cpp:62]: We will call `PrepareDBusConnection()` again at the top of the while-loop. Is that intended?

Comment (simonlingoogle) [reply src/dbus/server/dbus_agent.cpp]: Good catch! Failed to remove the call below.

Comment (simonlingoogle) [reply src/dbus/server/dbus_agent.cpp]: Another thought is that, thanks to `unique_ptr`, even such mistake won't cause DBus connection leak.

**Review by jwhui — APPROVED** [review]

**Pull request merged**